android-components/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt

```diff
@@ -93,6 +99,23 @@ abstract class PlacesStorage(
         readScope.coroutineContext.cancelChildren()
     }
 
+    /**
+     * Cleans up pending read operations of a specific query.
+     *
+     * @param queryToCancel Previous query to cancel reads for.
+     * Calling cancel multiple times for the same query has effect only the first time.
+     * Use this in scenarios where the same instance is used in multiple scenarios to prevent cases
+     * in which a general cancel operation for one scenario cancels other reads for the same query.
+     * If the value is an empty string all current reads are immediately cancelled.
+     */
+    override fun cancelReadsOfQuery(queryToCancel: String) {
```

Comment:
Driveby - Maybe we can rename this to `cancelQueryReads`?

Reply:
🔍 nit: I think we can also just call this `cancelReads(nextQuery: String)`